src/components/Buttons/PrimaryButton.jsx

```diff
@@ -0,0 +1,29 @@
+import React from 'react'
+import PropTypes from 'prop-types'
+import { Button } from 'cozy-ui/react'
+
+const PrimaryButton = ({ className, onClick, icon, children, ...props }) => (
+  <div>
+    <Button
+      className={className}
+      onClick={onClick}
+      label={children}
+      icon={icon}
+      {...props}
+    />
+  </div>
+)
+
+PrimaryButton.propTypes = {
+  className: PropTypes.string,
+  onClick: PropTypes.func,
+  icon: PropTypes.any,
+  children: PropTypes.string.isRequired
+}
+PrimaryButton.defaultProps = {
+  className: '',
+  onClick: () => {},
+  icon: null
+}
+
+export default PrimaryButton
```

Comment:
Thanks for your question.

How to handle buttons in a toolbar? is the question I try to address:

__within html__
You create a button element `<button id="mybutton" class="btn btn-primary">Click Me!</button>`, then you attach an event listener on it `document.querySelector("#mybutton").addEventListener("click", displayImportModal)` so that the user could display the import modal when he clicks on it.

__within components-based application__
You create an full-featured element `<DisplayImportModalButton />` that is responsible to create the markup including the text node, attach the event.
Or you don't create that layer of abstraction and you use the generic `Button` element: `<Button className="btn" onClick={displayImportModal} label="Click Me! />`. And without a central state management system, that is the only solution for the button to be aware of the event handler. Keeping this in mind and going one step further, create another button with slightly differences: `<Button className="btn" theme="secondary" onClick={displayImportModal} label="Click Me Too!" />`. Now we see two buttons with a lot of code in common, and worse with useless code that obstructs code reading. That is the time for abstraction: put the constant in one place and declare the API for your application needs.

If the `cozy/cozy-ui` button element API is updated, you, as the developper of `cozy/cozy-contacts` have one and only place to go to update the coupling between your application and the library you depend on. This is a layer of abstracton that decouples the differents elements (notice on how many files you can find a reference to `"cozy-ui/react/Button"`)

_I just realize there are other places that need refactoring to remove the strong coupling between the app elements and the underlying dependencies. Rome wasn't built in a day._

_Note: HTML gives a bunch of elements with their API. `<button />` is one of them and its API to display the text node is to use composition `<button>Click Me!</button>`. I don't know why the choice have been made to be different and to use a props instead of composition._

Reply:
Ok, thanks for the explanation!

> You create an full-featured element <DisplayImportModalButton /> that is responsible to create the markup including the text node, attach the event.

I'm on board with that, so `ImportGoogleButton`, `ImportVcardButton` etc make sense to me and are 👍 . But I don't think the `PrimaryButton` abstraction below is helpful.

> Now we see two buttons with a lot of code in common

I mean not really, in that example the only real common part was the class. And in the cozy-contact code it's pretty much the same, some props might be in common here and there, but the main ones (`onClick`, `label`, `icon`, etc) are still pretty much unique. And in fact `PrimaryButton` takes pretty much the same props as `Button`, so there's no real win using `PrimaryButton` instead of just `Button`.

> If the cozy/cozy-ui button element API is updated, you, as the developper of cozy/cozy-contacts have one and only place to go to update the coupling between your application and the library you depend on.

That's a fair point, but the same thing can be said about any part of code that comes from a third party. We could wrap `Modals`, `Navs` or even React APIs and whatnot. So the overhead worth it?  
Sometimes yes, when we know we might swap out a library, or if an API is in beta, etc. But the `Button` API is quite stable by now, and we have control over it — the chances that a breaking change occurs is low enough that I don't think we need this abstraction.

Reply:
If you say so. A commit to ~~update~~ revert that is on its way.